src/lib.rs

```diff
@@ -326,19 +323,18 @@ pub trait BatchDomain: Domain {
 }
 
 /// Trait that provides access to the components of an encrypted transaction output.
-///
-/// Implementations of this trait are required to define the length of their ciphertext
-/// field. In order to use the trial decryption APIs in this crate, the length must be
-/// either [`ENC_CIPHERTEXT_SIZE`] or [`COMPACT_NOTE_SIZE`].
-pub trait ShieldedOutput<D: Domain, const CIPHERTEXT_SIZE: usize> {
+pub trait ShieldedOutput<D: Domain> {
     /// Exposes the `ephemeral_key` field of the output.
     fn ephemeral_key(&self) -> EphemeralKeyBytes;
 
     /// Exposes the `cmu_bytes` or `cmx_bytes` field of the output.
     fn cmstar_bytes(&self) -> D::ExtractedCommitmentBytes;
 
-    /// Exposes the note ciphertext of the output.
-    fn enc_ciphertext(&self) -> &[u8; CIPHERTEXT_SIZE];
+    /// Exposes the note ciphertext of the output. Returns `None` if the output is compact.
+    fn enc_ciphertext(&self) -> Option<D::NoteCiphertextBytes>;
+
+    /// Exposes the compact note ciphertext of the output.
+    fn enc_ciphertext_compact(&self) -> D::CompactNoteCiphertextBytes;
```

Comment:
Likewise, we should avoid forcibly cloning here, and instead have a `Clone` bound on `Domain::CompactNoteCiphertextBytes`:
```suggestion
    /// Exposes the compact view of the output's note ciphertext.
    fn enc_ciphertext_compact(&self) -> &D::CompactNoteCiphertextBytes;
```

Also, document this change to the trait in the changelog.

Reply:
After discussion in the ZSA meeting, we'll leave this method as-is; the issue is that for non-compact actions, we can't get a reference to a fixed-length sub-array. The alternative would be to have a separate `CompactShieldedOutput` trait (which is effectively what we had in the past), but we can consider that refactor later.

Reply:
Got it: left `enc_ciphertext_compact` as-is.